web-registry/src/hooks/projects/useProjectsWithOrders.ts

```diff
@@ -0,0 +1,101 @@
+import { SellOrderInfo } from '@regen-network/api/lib/generated/regen/ecocredit/marketplace/v1/query';
+import {
+  ProjectInfo,
+  QueryProjectsResponse,
+} from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+import shuffle from 'lodash/shuffle';
+
+import {
+  ProjectsSellOrders,
+  useProjectsSellOrders,
+} from 'pages/Projects/hooks/useProjectsSellOrders';
+import useEcocreditQuery from 'hooks/useEcocreditQuery';
+import { useQuerySellOrders } from 'hooks/useQuerySellOrders';
+
+const hasMetadata = (project: ProjectInfo): boolean => !!project.metadata;
+
+const hasSellOrders = (
+  project: ProjectInfo,
+  sellOrders: SellOrderInfo[],
```

Comment:
Could all these utils functions use an argument object param `{project, sellOrders}` instead of named parameters? It will be easier to add/remove parameters in the future this way.

Reply:
mmm, I don't think it's necessary, really. These functions are small utilities, they shouldn't do anything else I think.

Reply:
mm, applying to the main selection function :+1:

Reply:
Since these functions won't probably be reused elsewhere it might be fine. But in general, I'm advocating for using object param by default. Recently I had to do some refactoring on a function without an object parameter and I had to change all function calls because the signature changed even if it was with an optional param, so let's try to avoid that if we can.

Reply:
Ok, I agree